agents/agents-tools/src/commonMain/kotlin/ai/koog/agents/file/tools/ReadFileTool.kt

```diff
@@ -0,0 +1,149 @@
+     *
+     * @property path absolute filesystem path to the target file
+     * @property startLine the first line to include (0-based, inclusive), defaults to 0
+     * @property endLine the first line to exclude (0-based, exclusive), -1 means read to end,
+     *   defaults to -1
+     */
+    @Serializable
+    public data class Args(
+        val path: String,
+        val startLine: Int = 0,
+        val endLine: Int = -1,
+    ) : ToolArgs
+
+    /**
+     * Contains the successfully read file with its metadata and extracted content.
+     *
+     * The result encapsulates a [FileSystemEntry.File] which includes:
+     * - File metadata (path, name, extension, size, content type, hidden status)
+     * - Content as either full text or line-range excerpt
+     *
+     * @property file the file entry containing metadata and content
+     */
+    @Serializable
+    public data class Result(val file: FileSystemEntry.File) : ToolResult.JSONSerializable<Result> {
+        override fun getSerializer(): KSerializer<Result> = serializer()
+
+        /**
+         * Converts the result to a structured text representation.
+         *
+         * Renders the file information in the following format:
+         * - File path with metadata in parentheses (size, line count if available, "hidden" if the file is hidden)
+         * - Content section with either:
+         *     - Full text for complete file reads
+         *     - Excerpt with line ranges for partial reads
+         *     - No content section if content is [FileSystemEntry.File.Content.None]
+         *
+         * @return formatted text representation of the file
+         */
+        override fun toStringDefault(): String = text { file(file) }
+    }
+
+    override val argsSerializer: KSerializer<Args> = Args.serializer()
+    override val descriptor: ToolDescriptor = Companion.descriptor
+
+    /**
+     * Reads file content from the filesystem with optional line range filtering.
+     *
+     * Performs validation before reading:
+     * - Verifies the path exists in the filesystem
+     * - Confirms the path points to a file (not a directory)
+     *
+     * @param args arguments specifying the file path and optional line range
+     * @return [Result] containing the file with its content and metadata
+     * @throws [ToolException.ValidationFailure] if the file doesn't exist, is a directory, or
+     *   cannot be read
+     * @throws [IllegalArgumentException] if line range parameters are invalid
+     */
+    override suspend fun execute(args: Args): Result {
+        val path = fs.fromAbsolutePathString(args.path)
+
+        validate(fs.exists(path)) { "File does not exist: ${args.path}" }
+        validate(fs.metadata(path)?.type == FileMetadata.FileType.File) {
+            "Path must point to a file, not a directory: ${args.path}"
+        }
+
+        val file = FileSystemEntry.File.of(
+            path,
+            content = FileSystemEntry.File.Content.of(
```

Comment:
possible issue: a content should not be read if a file is not a text file (see fs.getContentType)